### PR TITLE
HITL - Add feedback screen.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset adds a feedback screen to `rearrange_v2`.

This is implemented as an app state that is activated when finishing an episode. See the following changes for implementation details of the systems this is built with:
* https://github.com/facebookresearch/habitat-lab/pull/1965
* https://github.com/facebookresearch/habitat-lab/pull/2041

https://github.com/user-attachments/assets/19c29de7-618f-40df-b870-8c8ec28dbdae

https://github.com/user-attachments/assets/de8a87a4-7970-4ac8-b2e5-fa569c276f1f

## How Has This Been Tested

This changeset adds a screen that shows task success and feedback upon completing an episode.

This is implemented as an `AppStateBase` and the new UI system. See these PRs for implementation details:
* https://github.com/facebookresearch/habitat-lab/pull/1965
* https://github.com/facebookresearch/habitat-lab/pull/2041

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/rearrange_v2/app_state_feedback.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+from habitat_hitl.core.user_mask import Mask
+
+OK_BUTTON_ID = "feedback_ok"
+SCREEN_TIMEOUT = 30.0
+
+FONT_SIZE_LARGE: Final[int] = 32
+FONT_SIZE_SMALL: Final[int] = 24
+
+
+class AppStateFeedback(AppStateBase):
```
Comment (0mdc) [inline examples/hitl/rearrange_v2/app_state_feedback.py:31]: @zephirefaith We can use this state in place of `app_state_skip_episode` in `phase3`.

File: examples/hitl/rearrange_v2/app_state_feedback.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        self._app_service._client_message_manager.update_camera_transform(
+            cam_matrix, destination_mask=Mask.ALL
+        )
+
+        # Time limit to assess feedback.
+        self._elapsed_time += dt
+        remaining_time = SCREEN_TIMEOUT - self._elapsed_time
+        if remaining_time <= 0:
+            self._timeout = True
+            return
```
Comment (zephirefaith) [inline examples/hitl/rearrange_v2/app_state_feedback.py:89]: You're handling timeout here, right? Remove the TODO next to `self._timeout`?

**Review by zephirefaith — APPROVED** [review]
Look good and makes sense. Thank you! Left a couple nit comments.

Also getting the task explanation on-the-fly is superneat.

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 3]
```diff
@@ -524,12 +525,26 @@ def get_next_state(self) -> Optional[AppStateBase]:
...
             )
         elif self._is_episode_finished():
-            return create_app_state_load_episode(
-                self._app_service, self._app_data, self._session
-            )
+            success = self._metrics.get_task_percent_complete()
+            feedback = self._metrics.get_task_explanation()
+
+            # If task metrics are available, show task feedback.
+            if success is not None and feedback is not None:
```
[the remark this answers is not in the document]
Comment (0mdc) [reply examples/hitl/rearrange_v2/rearrange_v2.py:535]: Perhaps.

This is tweaked for running the current experiments, but it's possible that this interface is used with another experiment that doesn't provide the same metrics.

**Pull request merged**